src/server/pfs/server/driver.go

```diff
@@ -2377,89 +2443,194 @@ func nodeToFileInfo(commit *pfs.Commit, path string, node *hashtree.NodeProto, f
 	return fileInfo
 }
 
-func (d *driver) inspectFile(ctx context.Context, file *pfs.File) (*pfs.FileInfo, error) {
+func (d *driver) inspectFile(ctx context.Context, file *pfs.File) (fi *pfs.FileInfo, retErr error) {
 	pachClient := d.getPachClient(ctx)
 	ctx = pachClient.Ctx()
 	if err := d.checkIsAuthorized(pachClient, file.Commit.Repo, auth.Scope_READER); err != nil {
 		return nil, err
 	}
-	tree, err := d.getTreeForFile(ctx, file)
+	commitInfo, err := d.inspectCommit(ctx, file.Commit, pfs.CommitState_STARTED)
 	if err != nil {
 		return nil, err
 	}
-
-	node, err := tree.Get(file.Path)
+	// Handle commits to input repos
+	if commitInfo.Provenance == nil {
+		tree, err := d.getTreeForFile(ctx, file)
+		if err != nil {
+			return nil, err
+		}
+		node, err := tree.Get(file.Path)
+		if err != nil {
+			return nil, pfsserver.ErrFileNotFound{file}
+		}
+		return nodeToFileInfo(file.Commit, file.Path, node, true), nil
+	}
+	// Handle commits to output repos
+	if commitInfo.Finished == nil {
+		return nil, fmt.Errorf("output commit %v not finished", commitInfo.Commit.ID)
+	}
+	if commitInfo.Trees == nil {
+		return nil, fmt.Errorf("no file(s) found that match %v", file.Path)
+	}
+	rs, cleanup, err := d.getTree(ctx, commitInfo, file.Path)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := cleanup(); err != nil && retErr != nil {
+			retErr = err
+		}
+	}()
+	node, err := hashtree.Get(rs, file.Path)
 	if err != nil {
 		return nil, pfsserver.ErrFileNotFound{file}
 	}
-
-	// TODO(bryce) Add file path cleaning to this return
 	return nodeToFileInfo(file.Commit, file.Path, node, true), nil
 }
 
-func (d *driver) listFile(ctx context.Context, file *pfs.File, full bool) ([]*pfs.FileInfo, error) {
+func (d *driver) listFile(ctx context.Context, file *pfs.File, full bool, f func(*pfs.FileInfo) error) (retErr error) {
 	pachClient := d.getPachClient(ctx)
 	ctx = pachClient.Ctx()
 	if err := d.checkIsAuthorized(pachClient, file.Commit.Repo, auth.Scope_READER); err != nil {
-		return nil, err
+		return err
 	}
-
-	tree, err := d.getTreeForFile(ctx, client.NewFile(file.Commit.Repo.Name, file.Commit.ID, ""))
+	commitInfo, err := d.inspectCommit(ctx, file.Commit, pfs.CommitState_STARTED)
 	if err != nil {
-		return nil, err
+		return err
 	}
-
-	rootPaths, err := tree.Glob(file.Path)
+	g, err := globlib.Compile(file.Path, '/')
 	if err != nil {
-		return nil, err
+		// TODO this should be a MalformedGlob error like the hashtree returns
```

Comment:
Would this be easy to get in?

Reply:
Yeah, doing it.

Reply:
So doing this is a bit more weird than I expected. It does not really make sense to generate hashtree package errors outside of the hashtree package and there is probably a better way to accomplish this functionality entirely in the hashtree package. I think I am going to leave it as is for now.

Reply:
We'll probably want to refactor to have `MalformedGlob` just be it's own global type, maybe in a `globutil` pkg that this and `hashtree` can import. Feel free to not do this for now though.